_stbt/grid.py

```diff
@@ -140,63 +215,70 @@ def position_to_region(self, position):
             raise IndexError("Index out of range: position %r in %r" %
                              (position, self))
 
-    def region_to_index(self, region):
-        """Like `Grid.region_to_position`, but returns a 1D index instead of
-        a 2D position.
 
-        :param stbt.Region region: See `Grid.region_to_position`.
-        :returns: See `Grid.position_to_index`.
-        """
-        pos = self.region_to_position(region)
-        return pos.x + pos.y * self.cols
+class _GridIter(object):
+    def __init__(self, grid):
+        self.grid = grid
+        self.index = 0
 
-    def index_to_region(self, index):
-        """Like `Grid.position_to_region`, but takes a 1D index instead of a 2D
-        position.
+    def __iter__(self):
+        return self
 
-        :param int index: See `Grid.index_to_position`.
-        :returns: See `Grid.position_to_region`.
-        """
-        return self.position_to_region(self.index_to_position(index))
+    def __next__(self):
+        i = self.index
+        if i == self.grid.area:
+            raise StopIteration
```

Comment:
```suggestion
            raise StopIteration()
```

Reply:
The Python tutorial here uses `StopIteration` without the parentheses, in both the Python 2 & Python 3 versions of the doc:

https://docs.python.org/3/tutorial/classes.html#iterators

Reply:
I did a quick test and calling `next` on a generator function does raise an **instance** of `StopIteration`, so I'll commit your suggestion.